### PR TITLE
QUARKUS-1408 Dev Service for MS SQL Client

### PR DESCRIPTION
**Pull request by jsmrcka**
https://issues.redhat.com/browse/QUARKUS-1408

Integration of the reactive MS SQL client with Dev Services.

Guides:
- https://quarkus.io/version/main/guides/datasource#reactive-datasource-2
- https://quarkus.io/version/main/guides/datasource#dev-services
- https://quarkus.io/version/main/guides/reactive-sql-clients

### CODE REVIEW

File: QUARKUS-1408.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Dev Service for MS SQL Client
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1408
+
+Related test plan: [Quarkus-959 - Improved Developer Joy with DevServices](https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-959.md)  
+Related test development: [Reactive PostGreSQL and MySQL Dev Services ](https://issues.redhat.com/browse/QUARKUS-1080)
```
Comment (rsvoboda) [inline QUARKUS-1408.md:6]:
```suggestion
Related test development: [Reactive PostgreSQL and MySQL Dev Services ](https://issues.redhat.com/browse/QUARKUS-1080)  
```
PostgreSQL is the official name

File: QUARKUS-1408.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Dev Service for MS SQL Client
```
Comment (rsvoboda) [inline QUARKUS-1408.md:1]:
```suggestion
# QUARKUS-1408 - Dev Service for MS SQL Client
```

File: QUARKUS-1408.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Test in Quarkus upstream:
+- https://github.com/quarkusio/quarkus/tree/main/extensions/reactive-mssql-client/deployment/src/test/java/io/quarkus/reactive/mssql/client
+
+## Automated test development
+In QE TS, extend the `sql-db/reactive-vanilla` with `mssql` datasource and separate MS SQL scenario:
+- Test that datasource is operational.
+- Test that Dev Services correctly initialize the MS SQL container.
+
+## Impact on test suites and test environment
+1 new scenario in existing module, additional pull and start of MS SQL image. Estimated time increase: 1 minute.
```
Comment (rsvoboda) [inline QUARKUS-1408.md:30]: Native will take longer.
```suggestion
1 new scenario in existing module, additional pull and start of MS SQL image.
Estimated time increase:
 - 1 minute for JVM mode
 - 5 minutes for Native mode
```

Side question: Is OpenShift also covered in dev mode / remote dev mode tests?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Comment by jsmrcka** [comment]
Added commit containig fixes for other TPs (https://github.com/quarkus-qe/quarkus-test-plans/pull/63/commits/ff790b8e16fe71f04f2629567af208df60c13b15).

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**